### DISCUSSION TITLE
mobile app with nextjs app route

### DISCUSSION THREAD

**Discussion by ruru-m07** [vercel/community#6246] [Help] [answered]
dose it possible to create a mobile app with the power of nextjs 14 app route.

like expo and react native

**Comment by junaidMJA** [comment]
No, Next.js is a React.js Framework used to build Web Applications only.

**Reply by ruru-m07** [reply]
@junaidMJA soo it will possible in future to create native app with power of nextjs what u think?

**Reply by junaidMJA** [reply]
There's nothing such as of now, unless React(Meta) allows creating mobile apps using React.js, Next.js won't be able to power mobile apps and this ain't happening because React already have React Native for this. But, yes if Next.js (Vercel) decides to create a Framework for React Native with similar functionalities and features.

**Reply by ruru-m07** [reply]
@junaidMJA glad to see the progress of vercel it looks like in future vercel should create something like that 🔥.

**Reply by junaidMJA** [reply]
As said by @versecafe you can use next-pwa developed by vercel to create a PWA in Next.js, they work like a native app but need a browser to work on.

**Comment by versecafe** [comment, marked as answer]
@ruru-m07 You can use something like [PWA Builder](https://www.pwabuilder.com/) to build native web view wrappers around a PWA for the play & app stores, and you can set a PWA for NextJS with either [next-pwa](https://www.pwabuilder.com/) which is outdated but very simple to configure, or with [serwist](https://serwist.pages.dev/) for an actively maintained system.

**Reply by ruru-m07** [reply]
@versecafe your information is pretty nice thanks ❤️ but i need more deeper information.

**Reply by versecafe** [reply]
Essentially you would be building a PWA in a web view so no access to native features except through the browser API, and you get to have fun with font size < 16 for inputs on iOS doing a weird zoom, your other option is going about it from the other direction using React Native with Expo and then [next-adapter](https://docs.expo.dev/guides/using-nextjs/#limitations-or-differences-compared-to-the-default-expo-for-web) to generate a NextJS application from react native but it is limited to the pages directory and relatively experimental.
